crates/bevy_render/macros/src/as_bind_group.rs

```diff
@@ -0,0 +1,389 @@
+                                    resource: bindings[#binding_vec_index].get_binding(),
+                                }
+                            });
+                            BindingState::Occupied {
+                            binding_type,
+                            ident: field_name,
+                        }},
+                    }
+                },
+                BindingState::Occupied { binding_type, ident: occupied_ident} => panic!(
+                    "The '{field_name}' field cannot be assigned to binding {binding_index} because it is already occupied by the field '{occupied_ident}' of type {binding_type:?}."
+                ),
+                BindingState::OccupiedConvertedUniform => panic!(
+                    "The '{field_name}' field cannot be assigned to binding {binding_index} because it is already occupied by a struct-level uniform binding at the same index."
+                ),
+                BindingState::OccupiedMergableUniform { uniform_fields } => {
+                    match binding_type {
+                        BindingType::Uniform => {
+                            uniform_fields.push(field);
+                        },
+                        _ => {panic!("The '{field_name}' field cannot be assigned to binding {binding_index} because it is already occupied by a {:?}.", BindingType::Uniform)},
+                    }
+                },
+            }
+
+            match binding_type {
+                BindingType::Uniform => { /* uniform codegen is deferred to account for combined uniform bindings */
+                }
+                BindingType::Texture => {
+                    binding_impls.push(quote! {
+                        #render_path::render_resource::OwnedBindingResource::TextureView({
+                            let handle: Option<&#asset_path::Handle<#render_path::texture::Image>> = (&self.#field_name).into();
+                            if let Some(handle) = handle {
+                                images.get(handle).ok_or_else(|| #render_path::render_resource::AsBindGroupError::RetryNextUpdate)?.texture_view.clone()
+                            } else {
+                                fallback_image.texture_view.clone()
+                            }
+                        })
+                    });
+
+                    binding_layouts.push(quote!{
+                        #render_path::render_resource::BindGroupLayoutEntry {
+                            binding: #binding_index,
+                            visibility: #render_path::render_resource::ShaderStages::all(),
+                            ty: #render_path::render_resource::BindingType::Texture {
+                                multisampled: false,
+                                sample_type: #render_path::render_resource::TextureSampleType::Float { filterable: true },
+                                view_dimension: #render_path::render_resource::TextureViewDimension::D2,
```

Comment:
We will need to support different texture / texture view dimensions, as you noted.

Reply:
Agreed!

Reply:
(resolving for now though as this isn't required functionality)